### PR TITLE
Tracer for reclosure of RAMP closed appeals job

### PR DESCRIPTION
**Pull request by shanear**
connects #5786

The plan here is to merge the code that just queries for appeals that we'd mark for reclosure and report to slack what they are. When we verify that that number is correct, we can go ahead and add the code to actually close them.

### CODE REVIEW

File: spec/models/ramp_closed_appeal_spec.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        RampClosedAppeal.create!(vacols_id: "SHANE1"),
+        RampClosedAppeal.create!(vacols_id: "SHANE2"),
+        RampClosedAppeal.create!(vacols_id: "SHANE3")
+      ]
+    end
+
+    before do
+      expect(AppealRepository).to receive(:find_ramp_reopened_appeals)
+        .with(%w[SHANE1 SHANE2 SHANE3])
+        .and_return(["STUFF"])
```
Comment (leikkisa) [inline spec/models/ramp_closed_appeal_spec.rb]: Haha

File: app/models/ramp_closed_appeal.rb [hunk 2]
```diff
@@ -2,4 +2,15 @@ class RampClosedAppeal < ApplicationRecord
   belongs_to :ramp_election
 
   delegate :established_at, to: :ramp_election
+
+  def self.reclose_all!
+    appeals_to_reclose = []
+
+    find_in_batches(batch_size: 800) do |batch|
+      appeals_to_reclose += AppealRepository.find_ramp_reopened_appeals(batch.map(&:vacols_id))
```
Comment (leikkisa) [inline app/models/ramp_closed_appeal.rb]: I don't understand how reclose_all gets the vacols_id but I think that's just me! Also, I have no idea how I turned this yellow :-P

Comment (cmgiven) [reply app/models/ramp_closed_appeal.rb]: Check out: http://api.rubyonrails.org/classes/ActiveRecord/Batches.html#method-i-find_in_batches

Did not know this existed, but pretty handy.

Shane, is there a reason to drop batch size to 800? Confirmed that Oracle supports lists of up to 1,000 elements.

File: app/services/appeal_repository.rb [hunk 3]
```diff
@@ -274,6 +274,19 @@ def self.location_after_dispatch(appeal:)
     "98"
   end
 
+  # Finds appeals in the set of vacols_ids passed that have been reopened after
+  # being closed for RAMP
+  def self.find_ramp_reopened_appeals(vacols_ids)
+    VACOLS::Case
+      .where(bfkey: vacols_ids)
+      .where.not(bfdc: "P")
```
Comment (cmgiven) [inline app/services/appeal_repository.rb]: This line is duplicative of 284.

File: app/models/ramp_closed_appeal.rb (hunk 2 above)
Comment (shanear) [reply app/models/ramp_closed_appeal.rb]: Just to be safe i guess. It doesn't really slow it down much

File: app/services/appeal_repository.rb (hunk 3 above)
Comment (shanear) [reply app/services/appeal_repository.rb]: good catch

**Pull request merged**